lib/core/src/lib/form/components/widgets/core/form-field.model.spec.ts

```diff
@@ -501,41 +530,112 @@ describe('FormFieldModel', () => {
+                ]);
+            });
+
+            it('object with id, name and options', () => {
+                const field = new FormFieldModel(new FormModel(), {
+                    type: FormFieldTypes.RADIO_BUTTONS,
+                    options: [{ id: 'lower-priority-opt', name: 'Lower Priority Option' }],
+                    value: {
+                        id: 'opt2',
+                        name: 'Option 2',
+                        options: [
+                            { id: 'opt1', name: 'Option 1' },
+                            { id: 'opt2', name: 'Option 2' }
+                        ]
+                    }
+                });
+
+                expect(field.value).toBe('opt2');
+                expect(field.options).toEqual([
+                    { id: 'opt1', name: 'Option 1' },
+                    { id: 'opt2', name: 'Option 2' }
+                ]);
+            });
         });
-        expect(field.hasMultipleValues).toBe(true);
-        expect(field.hasEmptyValue).toBe(false);
-    });
 
-    it('should parse and resolve radio button value', () => {
-        const field = new FormFieldModel(new FormModel(), {
-            type: FormFieldTypes.RADIO_BUTTONS,
-            options: [
-                { id: 'opt1', name: 'Option 1' },
-                { id: 'opt2', name: 'Option 2' }
-            ],
-            value: 'opt2'
+        describe('should leave not resolved value (in case of delayed options)', () => {
+            it('when string', () => {
+                const field = new FormFieldModel(new FormModel(), {
+                    type: FormFieldTypes.RADIO_BUTTONS,
+                    options: [],
+                    value: 'delayed-option-id'
+                });
+
+                expect(field.value).toBe('delayed-option-id');
+            });
+
+            it('when object', () => {
+                const field = new FormFieldModel(new FormModel(), {
+                    type: FormFieldTypes.RADIO_BUTTONS,
+                    options: [],
+                    value: { id: 'delayed-option-id', name: 'Delayed option' }
+                });
+                expect(field.value).toBe('delayed-option-id');
+            });
         });
 
-        expect(field.value).toBe('opt2');
-    });
+        it('should add value (selected option) to field options if NOT present', () => {
+            const field = new FormFieldModel(new FormModel(), {
+                type: FormFieldTypes.RADIO_BUTTONS,
+                options: [],
+                value: { id: 'opt1', name: 'Option 1' }
+            });
 
-    it('should parse and leave radio button value as is', () => {
-        const field = new FormFieldModel(new FormModel(), {
-            type: FormFieldTypes.RADIO_BUTTONS,
-            options: [],
-            value: 'deferred-radio'
+            expect(field.options).toEqual([{ id: 'opt1', name: 'Option 1' }]);
+            expect(field.value).toEqual('opt1');
         });
```

Comment:
Similar what we do for dropdown

Reply:
When we will preview completed forms with radio buttons from older projects we will see at least selected option, the newly created will store full list of options